### PR TITLE
feat: support seeing who liked a message

### PR DESCRIPTION
**Pull request by hydr063n**
fixes https://github.com/status-im/nim-status-client/issues/1386

Added a tooltip that shows username or alias of chat keys stored in the emojiReactions model fromAccount attribute

reaction should show who made them

### CODE REVIEW

**Review by jrainville — COMMENTED** [review]
This looks good, though I wonder if there are, let's say, 20 people putting reactions, would it look super goofy?
Maybe we need to elide after a point or something like that

**Comment by hydr063n** [comment]
Thanks! Thats true, it wont be very appealing if there are too many reactions. I have added support for eliding after 5 authors just like discord.

The algorithm works like this:
There are 2 nodes (left and right). The left node will "try" to get max 3 authors while the right node is filled with the remaining authors.
The goal is to show leftNode authors first (comma seperated) and then decide whether to show right node authors as:
1. the name of the 4th author or
2. the number of authors starting from the 4th and upwards

left and right nodes are joined with "and" instead of a comma.

This results in a tooltip with the following format
`jona`
`jona and rich`
`jona, rich and vitalik`
`jona, rich, vitalik and jenny`
`jona, rich, vitalik and 2 more reacted`
`jona, rich, vitalik and 3 more reacted`
.
`jona, rich, vitalik and n more reacted`

react-test1

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/EmojiReactions.qml [hunk 1]
```diff
@@ -8,6 +9,30 @@ Item {
     height: 20
     width: childrenRect.width
 
+    function lastTwoItems(left, right) {
+        return left + " and " + right;
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/EmojiReactions.qml:13]:
```suggestion
        return qsTr("%1 and  %2").arg(left, right);
```

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/EmojiReactions.qml [hunk 2]
```diff
@@ -8,6 +9,30 @@ Item {
...
+    }
+
+    function showReactionAuthors(fromAccounts) {
+        if (fromAccounts.length < 3)
+            return fromAccounts.join(" and ");
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/EmojiReactions.qml:18]:
```suggestion
            return fromAccounts.join(qsTr(" and "));
```

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/EmojiReactions.qml [hunk 3]
```diff
@@ -8,6 +9,30 @@ Item {
...
+
+        var leftNode = [];
+        var rightNode = [];
+
+        if (fromAccounts.length > 3) {
+            leftNode = fromAccounts.slice(0, 3);
+            rightNode = fromAccounts.slice(3, fromAccounts.length);
+            return (rightNode.length == 1) ?
+                                lastTwoItems(leftNode.join(", "), rightNode[0]) :
+                                lastTwoItems(leftNode.join(", "), `${rightNode.length} more reacted.`);
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/EmojiReactions.qml:28]:
```suggestion
                                lastTwoItems(leftNode.join(", "), qsTr("%1 more reacted.").arg(rightNode.length);
```

**Review by jrainville — APPROVED** [review]
Very nice. It looks clean. I just added the `qsTr` function to make sure those strings get translated

**Review by jrainville — APPROVED** [review]

**Pull request merged**